src/components/pages/Invite/SendInvite.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react";
+import { Button, TextInput, View } from "react-native";
+import { connect, MapStateToProps, MergeProps } from "react-redux";
+import { ApiEndpoint } from "../../../api";
+import { RahaState } from "../../../store";
+import { sendInvite } from "../../../store/actions/members";
+import {
+  ApiCallStatus,
+  ApiCallStatusType
+} from "../../../store/reducers/apiCalls";
+import { getStatusOfApiCall } from "../../../store/selectors/apiCalls";
+import { Text } from "../../shared/elements";
+import validator from "validator";
+
+type ReduxStateProps = {
+  sendInviteStatus?: ApiCallStatus;
+};
+
+type OwnProps = {
+  videoToken: string;
+};
+
+type SendInviteState = {
+  email?: string;
+  enteredInvalidEmail: boolean;
+};
+
+type SendInviteProps = OwnProps &
+  ReduxStateProps & {
+    sendInvite: (videoToken: string, email: string) => void;
+  };
+
+class SendInviteView extends React.Component<SendInviteProps, SendInviteState> {
+  constructor(props: SendInviteProps) {
+    super(props);
+    this.state = {
+      enteredInvalidEmail: false
+    };
+  }
+
+  sendInvite = () => {
+    const enteredEmail = this.state.email;
+    // TODO: Check if already a member
+    if (!enteredEmail || !validator.isEmail(enteredEmail)) {
+      this.setState({ enteredInvalidEmail: true });
+      return;
+    }
+    this.props.sendInvite(this.props.videoToken, enteredEmail);
+  };
+
+  private _renderSendingStatus = () => {
+    const statusType = this.props.sendInviteStatus
+      ? this.props.sendInviteStatus.status
+      : undefined;
+    switch (statusType) {
+      case ApiCallStatusType.STARTED:
+        return <Text>Sending invite...</Text>;
+      case ApiCallStatusType.SUCCESS:
+        return <Text>Invite successful!</Text>;
+      case ApiCallStatusType.FAILURE:
+        return <Text>Invite failed.</Text>;
+      default:
+        return undefined;
+    }
+  };
+
+  render() {
+    return (
+      <React.Fragment>
+        <TextInput
```

Comment:
I'd make it a [controlled component](https://reactjs.org/docs/forms.html#controlled-components) for state consistency, but I don't think trimming will cause any inconsistency issues so it's fine as is

Reply:
I'll look into it with the additional validation stuff!